WordPress/gutenberg
# Add compound class to layout wrapper for global spacing styles

## What?


In #45326 I found that adding spacing support to Cover block didn't work out of the box, due to Cover's nested markup structure. As of #44600, layout classnames are added to the block's inner wrapper, so the `.wp-block-cover.is-layout-flow` selector used for block spacing won't work.

Because spacing styles need to target only the direct descendants of the layout wrapper, simply separating the selectors e.g. `.wp-block-cover .is-layout-flow` also won't work. 

We need a selector that targets the layout wrapper for each particular block type. This PR implements that selector, and adds spacing support to the Cover block as a testable example.


Note: I deliberately avoided any solution that would require specifying a target wrapper classname in the layout object, because once #47807 is done it should be possible to enable/disable layout on any block from theme.json, without knowledge of the block's markup structure. Also, the whole point of  #44600 was to avoid doing precisely that 😄 



## Testing Instructions





1. In global styles, set a block spacing value for Cover block and another for Group block;
2. Add a Cover to a post or template, add a couple of Paragraphs and a Group block to it, then a couple more Paragraphs inside the Group block;
3. Check that the correct spacing values are used on both Cover and Group block children;
4. Set a different block spacing value in the Cover block settings and check that it overrides the global value.

### Testing Instructions for Keyboard


## Screenshots or screencast 

Front end with global spacing:

<img width="1455" alt="Screenshot 2023-02-10 at 3 01 08 pm" src="https://user-images.githubusercontent.com/8096000/217997604-d9010446-fb75-48b5-9827-c5ffec858f08.png">

Block editor settings:

<img width="1046" alt="Screenshot 2023-02-10 at 2 59 39 pm" src="https://user-images.githubusercontent.com/8096000/217997631-3c07c2c4-4e49-4d34-80b6-ada90bae6786.png">
